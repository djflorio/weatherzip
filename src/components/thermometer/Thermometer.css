.thermometer {
  position: relative;
  width: 100px;
  height: 20px;
  border-radius: 15px;
  overflow: hidden;
  border: solid 1px rgb(0, 92, 104);
}

.thermometer__temp {
  position: absolute;
  top: 2px;
  left: 10px;
  font-size: 0.8em;
}

.thermometer__fill {
  height: 100%;
  background-color: rgb(0, 179, 202);
}