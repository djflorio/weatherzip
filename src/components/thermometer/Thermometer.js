// Third party
import React from 'react';
import PropTypes from 'prop-types';

// Assets
import './Thermometer.css';


const Thermometer = (props) => {

  let width = props.temp;
  if (props.units === "metric") {
    // Convert scale to F for visual purposes only
    width = width * 1.8 + 32;
  }

  if (width < 0) {
    width = 0;
  } else if (width > 100) {
    width = 100;
  }

  const style = {
    width: width
  }

  return (
    <div className="thermometer">
      <span className="thermometer__temp">{Math.round(props.temp)}&deg;</span>
      <div className="thermometer__fill" style={style}></div>
    </div>
  );
}

Thermometer.propTypes = {
  units: PropTypes.string.isRequired,
  temp: PropTypes.number.isRequired
}

export default Thermometer;