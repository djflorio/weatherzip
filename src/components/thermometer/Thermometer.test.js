import Thermometer from './Thermometer';

it("renders correctly", () => {
  const wrapper = shallow(
    <Thermometer
      units="imperial"
      temp={70}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it("rounds the temperature", () => {
  const wrapper = shallow(
    <Thermometer
      units="imperial"
      temp={70.6}
    />
  );

  const text = wrapper.find(".thermometer__temp").text();

  expect(text).toEqual("71Â°");
});