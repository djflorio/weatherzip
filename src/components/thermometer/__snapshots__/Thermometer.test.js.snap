// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="thermometer"
>
  <span
    className="thermometer__temp"
  >
    70
    Â°
  </span>
  <div
    className="thermometer__fill"
    style={
      Object {
        "width": 70,
      }
    }
  />
</div>
`;
